### DISCUSSION TITLE
Sort Items of a query on the view

### DISCUSSION THREAD

**Discussion by MariaGanan** [2sic/2sxc#3290] [Q&A] [answered]
Hi, 
On an App I have a view with data from a query, is there a way to make this items be a list that I can sort or something?
I will like to be able to sort them like the content items, or simillar.
Thank you.

**Comment by iJungleboy** [comment]
I guess it depends on what you really want.

If you want the user to play around with the data, then something like https://datatables.net/ would probably help.

If you want to sort in in code, you should use LINQ

**Reply by MariaGanan** [reply]
Thank you!

**Reply by vmasanas** [reply]
Hi Daniel, the request here is how an admin/editor can force some specific order on the items.
For example, we have a list of documents and we need to present them in a very specific way. Not alphabetical, not by date.
And the editor of the documents should be able the order them according to whatever preference they need.

The idea is not that the end user (visitor) can sort the items, but that they come in some predefined way.

One option would be add "Sort Number" on the item and let them fill this field with incremental numbers (1,2,3...) but it's not very user friendly. 
The idea would be something similar to an up / down arrow on each item that sets this order. 
Or imagine a FAQs module where you need to order the items in a specific way.

Do you have any app that would do something like this?

**Reply by iJungleboy** [reply, marked as answer]
@vmasanas This is a very common use case 😉 - and of course 2sxc can do this

The most common way to do this is to create a view which shows the type you want to manually sort etc. and activate list. 

Then the user can 

1. create new items 
2. add existing ones
3. move them up down
4. enter a sub-dialog to reorder using drag-drop

You can find some examples in the standard content-app. For example, add the Link template, add some links, reorder etc.
